include/drivers/interrupt_controller/gic.h

```diff
@@ -252,9 +252,13 @@
 #define GIC_INTID_SPURIOUS		1023
 
 /* Fixme: update from platform specific define or dt */
-#define GIC_NUM_CPU_IF			1
+#define GIC_NUM_CPU_IF			CONFIG_MP_NUM_CPUS
```

Comment:
Question around SMP: we have a config flag for `SMP` and a config flag for `MP_NUM_CPUS`, which seems minorly redundant. SMP with one CPU doesn't make sense, and non-SMP with multiple CPUs doesn't seem to make sense either (especially given that e.g. curr_cpu returns cpus[0] in that case).

When should we use `CONFIG_SMP` versus `MP_NUM_CPUS > 1`?

Reply:
There is a test that not have SMP, but need mutilple cpus. "tests/kernel/mp"